### PR TITLE
This commit adds RR DNS support for multimaster mode

### PR DESCRIPTION
**Pull request by gczuczy**
### What does this PR do?
For multimaster mode this commit adds RR DNS support. SRV records are not supported by this PR currently, as SRVs don't make sense in multimaster mode, only with master_type==failover.

### What issues does this PR fix or reference?
Fixes: All the issues which mentioned RR DNS support for multimaster mode. SRV is doesn't make sense for multimaster mode, so it's not addressed in this commit.

### Previous Behavior
If the masters specified by a hostname were a round-robin DNS record, only the first resolved entry was returned and handled as the master address, the rest ignored.

### New Behavior
With master_type==str (multimaster mode), a simple round-robin DNS record (or multiples, in case masters is a list) can be utilized to maintain the lost of masters. This way the list of masters can be maintained in the DNS subsystem in multimaster mode, and changes are not affecting the default minion configuration in the deployment.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

I would like to ask for help regarding the docs/changelog and tests.

### Commits signed with GPG?
No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/utils/network.py [hunk 1]
```diff
@@ -2153,6 +2153,18 @@ def refresh_dns():
         # Exception raised loading the library, thus RES_INIT is not defined
         pass
 
+def dns_resolve_addresses(host):
+    '''
+    This function returns all the IP address that the host looks up to.
+    IPv6 addresses are only returned if the python interpreter has been
+    compiled with IPv6 support, because otherwise the returned structure
+    is effectively garbage.
+    '''
```
Comment (barbaricyawps) [inline salt/utils/network.py:2162]: Thanks for your PR, @gczuczy . You're off to a good start here, but we probably need more information in order to make this documentation more effective.

Earlier this year, I started working on a set of standards for module documentation but had to put the project on hold temporarily. However, looking at what my research says should go into effective documentation for functions, it looks like the documentation would be more helpful if you could include:

- In the description, provide at least one use case. You should provide more than one if this function is particularly complex (such as a lot of kwargs), but it doesn't look like it is to me.
- Indicate the datatype for each parameter (string, dict, etc.). Returns should be included in the parameters.
- Provide an example of return data.

Your warning about how the IPv6 addresses need to be compiled with an interpreter that has IPv6 support is good, but should probably be formatted as an admonition (a warning or tip box) in order to make it more obvious. Here's some more information about how to format warnings and tip boxes in rST: https://saltstack.gitlab.io/open/docs/docs-hub/topics/rst-guide.html#admonitions-tips-note-boxes-warnings

Comment (gczuczy) [reply salt/utils/network.py]: @barbaricyawps Should the return type be specified by PEP484? I don't see that anywhere in the code currently.

Comment (gczuczy) [reply salt/utils/network.py]: @barbaricyawps  Please see `8ce2431ace` for the docstring update, I hope this loves up to the upcoming standards.

**Comment by gczuczy** [comment]
Should I expose the `dns_resolve_addresses` (`salt.utils.network:2157`) function to jinja? I think it might be a good idea to do so, it can come useful in reasonable situations writing state files.

**Comment by gczuczy** [comment]
As I can see the CHANGELOG is automated, so there's no need to update that.

**Comment by dwoz** [comment]
Closing this due to inactivity. Anyone should feel free to re-open it if they want to see it through to the end in one release cycle.

@gczuczy If this is something you want to get into 3008.x please re-open and remove the draft state. Thanks.

**Pull request closed without merging**